ExploreASL_Initialize.m

```diff
@@ -697,111 +639,242 @@
+    defaultProcessArray = [0 0 0];
+    defaultSkipPause = 0;
+    defaultiWorker = 1;
+    defaultnWorkers = 1;
+    
+    % Add definitions to the input parser
+    addOptional(p, 'DataParPath', defaultDataParPath, validDataParPath);
+    addOptional(p, 'ImportArray', defaultImportArray, validImportArray);
+    addOptional(p, 'ProcessArray', defaultProcessArray, validProcessArray);
+    addOptional(p, 'SkipPause', defaultSkipPause, validSkipPause);
+    addOptional(p, 'iWorker', defaultiWorker, validiWorker);
+    addOptional(p, 'nWorkers', defaultnWorkers, validnWorkers);
+    
+    % Parse input
+    parse(p,varargin{:});
 
-for iField=1:length(field_symbol)
-    if ~isfield(x,field_symbol{iField})
-        warning(['x.' field_symbol{iField} ' was not defined in DATA_PAR.m!'])
-    end
 end
 
-if ~isfield(x,'D')
-    warning('x.D didn''nt exist');
-else
-    field_symbol = {'ROOT'};
-    for iField=1:length(field_symbol)
-        if ~isfield(x.D,field_symbol{iField})
-            warning(['x.D.' field_symbol{iField} ' was not defined in DATA_PAR.m!'])
-        end
+%% -----------------------------------------------------------------------
+%% Convert parsed input
+function parameters = convertParsedInput(parameters)
+
+    % Check if inputs are empty or chars
+    if isempty(parameters.DataParPath),     parameters.DataParPath = '';                                    end
+    if ischar(parameters.ImportArray),      parameters.ImportArray = str2num(parameters.ImportArray);       end
+    if ischar(parameters.ProcessArray),     parameters.ProcessArray = str2num(parameters.ProcessArray);     end
+    if ischar(parameters.SkipPause),        parameters.SkipPause = str2num(parameters.SkipPause);           end
+    if ischar(parameters.iWorker),          parameters.iWorker = str2num(parameters.iWorker);               end
+    if ischar(parameters.nWorkers),         parameters.nWorkers = str2num(parameters.nWorkers);             end
+    
+    % Check length of arrays (single digit input)
+    if length(parameters.ImportArray)<4
+        parameters.ImportArray = [parameters.ImportArray(1),...
+                                  parameters.ImportArray(1),...
+                                  parameters.ImportArray(1),...
+                                  parameters.ImportArray(1)];
+    end
+    if length(parameters.ProcessArray)<3
+        parameters.ProcessArray = [parameters.ProcessArray(1),...
+                                   parameters.ProcessArray(1),...
+                                   parameters.ProcessArray(1)];
+    end
+    
+    % Different default for deployed mode
+    if isdeployed
+        parameters.SkipPause = 0;
     end
+
+
 end
 
-if ~isempty(regexp(x.subject_regexp, '^(\^|)\.\*(\$|)$'))
-    warning('Subject regexp not specific! Check that no wrong folders are included as subjects');
+%% -----------------------------------------------------------------------
+%% Store parsed input
+function x = storeParsedInput(parameters)
```

Comment:
Jan and I got the habit to prefix subfunctions by their main function name, so this would be `ExploreASL_Initialize_storeParsedInput` in this case. This makes it nicely readable since this immediately makes it clear that this is a subfunction. If you agree I would apply this for all subfunctions in this function as well (to keep style same throughout)

Reply:
Done in 51f3f3b.